docs/OV_Runtime_UG/supported_plugins/GPU.md

```diff
@@ -230,16 +230,33 @@ For more details, see the :doc:`optimization guide<openvino_docs_deployment_opti
 Dynamic Shapes
 +++++++++++++++++++++++++++++++++++++++
 
-The GPU plugin supports dynamic shapes for batch dimension only (specified as ``N`` in the :doc:`layouts terms<openvino_docs_OV_UG_Layout_Overview>`) with a fixed upper bound. 
-Any other dynamic dimensions are unsupported. Internally, GPU plugin creates ``log2(N)`` (``N`` - is an upper bound for batch dimension here) 
-low-level execution graphs for batch sizes equal to powers of 2 to emulate dynamic behavior, so that incoming infer request 
-with a specific batch size is executed via a minimal combination of internal networks. For example, batch size 33 may be executed via 2 internal networks with batch size 32 and 1.
+GPU plugin supports dynamic shape and the general description can be found in :doc:`dynamic shapes guide<openvino_docs_OV_UG_DynamicShapes>`.
 
-.. note:: 
+To support dynamic shape execution, the following basic infrastructures are implemented:
+
+- Runtime shape inference : Infers the output shapes of each primitive for a new input shape at runtime
+- Shape agnostic kernels : New kernels that can run arbitrary shapes. If shape agnostic kernel is not available, the required kernel is compiled at runtime for each shape.
+- Asynchronous kernel compilation : Even when a shape agnostic kernel is available, the GPU plugin compiles an optimal kernerl for the given shape and preserve it in the in-memory cache for future use.
+- In-memory cache : Preserves kernels compiled in runtime and weights reordered for the specific kernels
+
+.. note::
+
+   Currently, dynamic shape support in GPU plugin is a preview feature and has following limitations:
+   - It mainly supports NLP models (Natural Language Processing). Not all operations and optimization passes supports dynamic shape. As a result, an arbitrary model may crash or experience significant performance drops.   
+   - Due to the dominant runtime overhead on the host device, dynamic shapes may perform worse than static shapes on a discrete GPU.
+   - Dynamic rank is not supported.
+
+Bounded dynamic batch
+-----------------------------------------------------------
 
-   Such approach requires much more memory and the overall model compilation time is significantly longer, compared to the static batch scenario.
+It is worth noting that the internal behavior differs from general cases in cases where only the batch dimension is dynamic and has a fixed upper bound, a.k.a bounded batch dynamic shape.
```

Comment:
nit: what about `from general cases in cases where` -> `from general cases if`?

Reply:
Applied